raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2276,6 +2277,24 @@ public void resign(int epoch) {
         );
     }
 
+    private void validateSnapshotId(OffsetAndEpoch snapshotId) {
```

Comment:
Can we make the comparisons with `snapshotId` consistently on the left-hand side?

E.g., 
```
snapshotId.offset > highWatermarkOpt.get().offset
```

Makes it a bit easier to parse this logic.

Reply:
Make sense, fixed